build/version.cake

```diff
@@ -0,0 +1,71 @@
+public class BuildVersion
+{
+    public string Version { get; private set; }
+    public string SemVersion { get; private set; }
+    public string Milestone { get; private set; }
+    public string CakeVersion { get; private set; }
+
+    public static BuildVersion CalculatingSemanticVersion(
+        ICakeContext context,
+        bool isLocalBuild,
+        bool isPublishBuild,
+        bool isReleaseBuild
+        )
+    {
+        if (context == null)
+        {
+            throw new ArgumentNullException("context");
+        }
+
+        string version = null;
+        string semVersion = null;
+        string milestone = null;
+
+        if (context.IsRunningOnWindows())
+        {
+            context.Information("Calculating Semantic Version");
+            if (!isLocalBuild || isPublishBuild || isReleaseBuild)
+            {
+                context.GitVersion(new GitVersionSettings{
+                    UpdateAssemblyInfoFilePath = "./src/SolutionInfo.cs",
+                    UpdateAssemblyInfo = true,
+                    OutputType = GitVersionOutput.BuildServer
+                });
+
+                version = context.EnvironmentVariable("GitVersion_MajorMinorPatch");
+                semVersion = context.EnvironmentVariable("GitVersion_LegacySemVerPadded");
+                milestone = string.Concat("v", version);
+            }
+
+            GitVersion assertedVersions = context.GitVersion(new GitVersionSettings
```

Comment:
@devlead I think we need a check here as well.  Running GitVersion again will only be required **if** we are unable to access the environment variables.